tests/core_tests/test_context.py

```diff
@@ -1137,22 +1170,90 @@ def test_dict_cleanup(self):
         for entity in self.kws["additional_entities"]:
             entity["created_at"] = datetime.datetime.now()
         self.kws["source_entity"]["created_at"] = datetime.datetime.now()
+        self.kws["user"]["created_at"] = datetime.datetime.now()
 
         expected = {
-            "additional_entities": [{"id": 42, "type": "Sequence"}],
-            "entity": {"code": "shot_name", "id": 2, "name": "shot_name", "type": "Shot"},
-            "project": {"id": 1, "name": "project_name", "type": "Project"},
-            "source_entity": {"id": 12, "type": "Version"},
-            "step": {"id": 4, "name": "step_name", "type": "Step"},
-            "task": {"id": 45, "type": "Task"},
-            "user": None
+            "task": {
+                "type": "Task",
+                "id": self.task["id"],
+                "name": "task_content"
+            },
+            "step": {
+                "type": "Step",
+                "id": self.step["id"],
+                "name": "step_name"
+            },
+            "entity": {
+                "type": "Shot",
+                "id": self.shot["id"],
+                "name": self.shot["code"]
+            },
+            "project": {
+                "type": "Project",
+                "id": self.project["id"],
+                "name": "project_name"
+            },
+            # Contrary to other entities, the source entity and...
+            "source_entity": {
+                "type": "Version",
+                "id": self.version["id"],
+            },
+            # ... additional entities' name are never resolved
+            # so they can be missing.
+            "additional_entities": [{
+                "type": "Sequence",
+                "id": self.seq["id"],
+            }],
+            "user": {
+                "type": "HumanUser",
+                "id": self.user["id"],
+                "name": "user_name"
+            }
         }
 
         ctx = context.Context(**self.kws)
+        pickled_data = ctx.serialize()
+
+        # Make sure there is only type and ids in the pickle.
+        unpickled_data = pickle.loads(six.ensure_binary(pickled_data))
+        for field in ["task", "step", "entity", "project", "source_entity", "user"]:
+            self.assertNotIn("name", unpickled_data[field])
+        self.assertNotIn("name", unpickled_data["additional_entities"][0])
 
         # Serialize/deserialize the object, we should have only kept type,
         # id and name-related fields.
-        ctx = context.deserialize(ctx.serialize())
+        real_get_field_from_row = self.mockgun._get_field_from_row
+
+        def _get_field_from_row_patch(entity_type, row, field):
+            value = real_get_field_from_row(entity_type, row, field)
+
+            # TODO: This fix should go inside mockgun. Linked fields in mockgun never
```

Comment:
Alright, there what these `Bad test data` comments come from!

There's a bug in Mockgun, where a linked entity field only returns the type and id of the linked entity and not it's name. Given that here we're unserializing a context and expecting to retrieve the name from Mockgun, this is a pretty bad bug.

So I originally went about and fixed Mockgun, but this introduced a lot of issues
1- some entities in mockgun has been created with the name in the wrong field or was missing entirely, so the linked entity field couldn't be set properly
2- doing so started introducing a lot of changes to the tests, as you've been in the comments above, but it started breaking a lot of other tests because there's a lot of very bad test data

In the end, it was proving to be way too costly to fix mockgun properly and then fix all the tests that were starting to break left and right, so I had to revert the mockgun changes and instead inject it here for the purpose of this test.